resources/js/components/shared/FilterTable.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <div class="pm-table-container">
+    <table class="pm-table-filter">
+      <thead>
+        <tr>
+          <th class="pm-table-border" :colspan="headers.length"></th>
+        </tr>
+        <tr>
+          <th
+            class="pm-table-ellipsis-column"
+            v-for="(column, index) in headers"
+            :key="index"
+          >
+            <div
+              class="pm-table-column-header"
+              :style="{ width: column.width + 'px' }"
+            >
+              <slot :name="column.field">
+                {{ column.label }}
+              </slot>
+            </div>
+            <div class="pm-table-filter-button">
+              <slot :name="`filter-${column.field}`">
+                
+              </slot>
+            </div>
+            <div
+              v-if="index !== headers.length - 1"
+              class="pm-table-column-resizer"
+              @mousedown="startResize(index)"
+            >
+            </div>
+          </th>
+        </tr>
+        <tr>
+          <th class="pm-table-border" :colspan="headers.length"></th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr
+          v-for="(row, rowIndex) in data.data"
+          :key="rowIndex"
+          @click="handleRowClick(row)"
+        >
+          <slot :name="`row-${rowIndex}`">
+          <td
+            v-for="(header, index) in headers"
+            :key="index"
+          >
+            <div v-if="containsHTML(row[header.field])" v-html="row[header.field]"></div>
+            <template v-else>
+              <template v-if="isComponent(row[header.field])">
+                <component 
+                  :is="row[header.field].component"
+                  v-bind="row[header.field].props"
+                >
+                </component>
+              </template>
+              <template v-else>
+                {{ row[header.field] }}
+              </template>
+            </template>
+          </td>
+          </slot>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  </template>
+
+<script>
+
+import moment from "moment";
+
+export default {
+  components: {
+  },
+  props: {
+    headers: [],
```

Comment:
add prop types and default values
example
` year: {
    default: 2016,
    type: Number
  }`